cql3/statements/create_view_statement.cc

```diff
@@ -175,6 +179,50 @@ std::pair<view_ptr, cql3::cql_warnings_vec> create_view_statement::prepare_view(
                 "before being replayed.", column_family(), _base_name.get_column_family()));
     }
 
+    const bool rf_rack_restricted_keyspaces = db.get_config().check_experimental(db::experimental_features_t::feature::RF_RACK_RESTRICTED_KEYSPACES);
+    // Technical note: `tools/schema_loader` uses this function, but does NOT provide an actually valid instance
+    //                 of `data_dictionary::database`. As a result, we may not have access to the underlying
+    //                 `locator::topology`, which is crucial for this logic.
+    //                 However, since in "normal" circumstances, i.e. when it's actually Scylla that calls this function,
+    //                 we should always be able to access this pointer. Hence this form of `if`.
+    if (rf_rack_restricted_keyspaces && ars.uses_tablets() && db.real_database_ptr()) {
+        // With tablets enabled, we want to restrict materialized views to keyspaces that satisfy RF == the number of racks OR RF == 1.
+        //
+        // Long story short, doing that, we want to ensure that:
+        // * pairing the base and view replicas occurs within the same rack,
+        // * there's only one replica per rack,
+        // * tablet replicas never leave their original rack.
+        //
+        // You can find more context about the pre-existing problems and solutions in: scylladb/scylladb#23030.
+        //
+        // Moreover, you can refer to the document "Consistency problems in Materialized Views arising from replica order changes"
+        // for an even broader perspective.
+
+        const locator::topology& topology = db.real_database_ptr()->get_token_metadata().get_topology();
+        // Note that a keyspace can only have tablets enabled if the used replication strategy is NetworkTopologyStrategy, cf.
+        //
+        // "When creating a new keyspace with tablets enabled (the default), you can still disable them on a per-keyspace basis.
+        //  The recommended NetworkTopologyStrategy for keyspaces remains required when using tablets."
+        //
+        // --- "Data Distribution with Tablets" in our documentation.
+        //
+        // That's why we're only considering that case and that's why this cast is correct.
```

Comment:
Maybe the cast is "correct", but I still think it would be nicer to check replication_strategy is not null (and use on_internal_error() if it is) than just allowing a crash if there is a loophole allowing the wrong strategy.

Imagine that in the future we invent a new SuperDuperRelicationStrategy and allow it with tablets, and we forget to fix this code and it can be used to crash Scylla...